src/toil/jobStores/aws/jobStore.py

```diff
@@ -1418,15 +1419,13 @@ def readFrom(self, readable):
 
                             if info.version is None:
                                 # Somehow we don't know the version. Try and get it.
-                                for attempt in retry_s3(predicate=lambda e: retryable_s3_errors(e) or isinstance(e, AssertionError)):
-                                    with attempt:
-                                        version = client.head_object(Bucket=bucket_name,
-                                                                     Key=compat_bytes(info.fileID),
-                                                                     **headerArgs).get('VersionId', None)
-                                        logger.warning('Loaded key for upload with no version and got version %s',
-                                                       str(version))
-                                        info.version = version
-                                        assert info.version is not None
+                                info.version = head_s3_object(
+                                    Bucket=bucket_name,
+                                    Key=compat_bytes(info.fileID),
+                                    **headerArgs
+                                ).get('VersionId', None)
+                                logger.warning('Loaded key for upload with no version and got version %s', str(info.version))
+                                assert info.version is not None
```

Comment:
We probably want to cut these; the warning shouldn't be a warning unless we got the wrong version result, and the assert probably should be a real `raise` if we actually want to stop if it isn't true in a live workflow.

Reply:
I agree.  This was code unchanged from the old function, and I was trying to make minimal changes.  I'll make that change though, I agree it's not necessary (and would like to get rid of versioned objects altogether in the future).